src/Dynatrace.OpenTelemetry.Exporter.Metrics.Tests/DynatraceMetricsExporterTests.cs

```diff
@@ -16,331 +16,1018 @@
+			var actualMetricString = await actualRequestMessage.Content!.ReadAsStringAsync();
+			Assert.Equal(expected, actualMetricString);
 
 			mockMessageHandler.Protected().Verify(
 				"SendAsync",
-				Times.Never(),
-				ItExpr.Is<HttpRequestMessage>(req => req.Method == HttpMethod.Post),
+				Times.Exactly(1),
+				ItExpr.IsAny<HttpRequestMessage>(),
 				ItExpr.IsAny<CancellationToken>());
 
+			AssertExportRequest(actualRequestMessage);
+
 			mockLogger.Verify(x => x.Log(
-				It.Is<LogLevel>(level => level == LogLevel.Warning),
-				It.IsAny<EventId>(),
-				It.Is<It.IsAnyType>((value, type) => value.ToString().Contains("Serialization")),
+				LogLevel.Warning,
+				7, // eventid define in LoggerExtensions
+				It.Is<It.IsAnyType>((value, type) =>
+					value.ToString()!.Contains("Received metric: 'double_counter' with cumulative aggregation temporality. Exporting as gauge")),
 				It.IsAny<Exception>(),
-				It.IsAny<Func<It.IsAnyType, Exception, string>>()), Times.Exactly(1));
+				It.IsAny<Func<It.IsAnyType, Exception, string>>()),
+				Times.Exactly(1));
 		}
 
-		private static List<Metric> CreateMetrics()
+		[Fact]
+		public async Task Export_DoubleSumDelta()
 		{
-			var metrics = new List<Metric>();
+			// Arrange
+			using var meter = new Meter(TestUtils.GetCurrentMethodName(), "0.0.1");
+
+			HttpRequestMessage actualRequestMessage = null!;
+			var mockMessageHandler = SetupHttpMock((HttpRequestMessage r) => actualRequestMessage = r);
+
+			var sut = new DynatraceMetricsExporter(null, null, new HttpClient(mockMessageHandler.Object));
+
+			// Delta is already the preferred temporality for our exporter
+			var metricReader = new TestMetricReader(sut);
+			using var provider = Sdk.CreateMeterProviderBuilder()
+				.AddMeter(meter.Name)
+				.AddReader(metricReader)
+				.Build();
+
+			var counter = meter.CreateCounter<double>("double_counter");
+			counter.Add(10.3, _attributes);
+
+			// Act - Reader will call our exporter
+			metricReader.Collect();
+
+			// Assert
+			var exportedMetrics = metricReader.GetExportedMetrics();
+			var point = MetricTest.FromMetricPoints(exportedMetrics.First().GetMetricPoints()).First();
 
-			var timestamp = DateTimeOffset.FromUnixTimeMilliseconds(1604660628881).UtcDateTime;
+			var expected = $"double_counter,attr1=v1,attr2=v2,dt.metrics.source=opentelemetry count,delta=10.3 {point.TimeStamp}";
+			var actualMetricString = await actualRequestMessage.Content!.ReadAsStringAsync();
+			Assert.Equal(expected, actualMetricString);
 
-			var metric1 = new Metric("namespace1", "metric1", "Description", AggregationType.LongSum);
-			metric1.Data.Add(new Int64SumData
+			mockMessageHandler.Protected().Verify(
+				"SendAsync",
+				Times.Exactly(1),
+				ItExpr.IsAny<HttpRequestMessage>(),
+				ItExpr.IsAny<CancellationToken>());
+
+			AssertExportRequest(actualRequestMessage);
+		}
+
+		[Fact]
+		public async Task Export_ObservableLongCounter()
```

Comment:
I think i have read, that observable long counters will never provide deltas (due to the way they function). Could you test what happens if you add to the counter (as is done here), then export, then add again, then export again? Will it export the cumulative value as deltas? or does that actually work?

Reply:
Actually it does export deltas 🤯.

I did this:

```csharp
var i = 1;
var observableCounter = meter.CreateObservableCounter("obs_counter", () =>
{
	return new List<Measurement<long>>()
	{
		new Measurement<long>(i++ * 10, _attributes)
	};
});
```
And when the metric reader is configured to export deltas: 

1st export
`obs_counter,attr1=v1,attr2=v2,dt.metrics.source=opentelemetry count,delta=10 1637169685051`

2nd export:
`obs_counter,attr1=v1,attr2=v2,dt.metrics.source=opentelemetry count,delta=10 1637169719650`

And when I change the reader to Cumulative:

1st export:
`obs_counter,attr1=v1,attr2=v2,dt.metrics.source=opentelemetry gauge,10 1637169525557`

2nd export
`obs_counter,attr1=v1,attr2=v2,dt.metrics.source=opentelemetry gauge,20 1637169605669`

Reply:
I added new tests for these scenarios:

Long:
`Export_ObservableLongCounterDelta`
`Export_ObservableLongCounterCumulative_ShouldExportAsGauge`

Double:
`Export_ObservableDoubleCounterDelta`
`Export_ObservableDoubleCounterCumulative_ShouldExportAsGauge`

Reply:
Update on this: I removed the cumulative tests as we now know for sure that the SDK will only provide Deltas because our exporter is configured to only support deltas via the `AggregationTemporality` attribute. The delta tests for ObservableCounters are still there.